docs/enhancements/1-crds-and-kubebuilder.md

```diff
@@ -0,0 +1,380 @@
+| Status | Authors   | Created    | 
+|--------|-----------|------------|
+| Review | @jewertow | Jan 2 2025 |
+
+# CRDs and kubebuilder
+
+## Overview
+
+We need CRDs to properly manage lifecycle of the Istio resources created by our controller.
+Currently, resources are not reconciled on user changes and are not removed when the controller is uninstalled, and that results in mesh misconfiguration.
```

Comment:
Another benefit here is that we can easily federate more meshes by providing external configuration (through CR) that does not require ctrl pod to restart, does not need error-prone cli flags as the only source of config, we can attach status to it without a need to parse events or logs to name a few. This is quite a UX improvement in my eyes.

Reply:
I rephrased the overview and included these arguments, thanks.